### PR TITLE
Add error handling to LDA page

### PR DESCRIPTION
**Pull request by vidyaap**

### CODE REVIEW

**Comment by davidatbu** [comment]
Thanks @vidyaap . I got approval to setup a dev server, so I'll do that and then hopefully test this PR over there :) Expect a review this afternoon.

**Comment by monajalal** [comment]
I will review after @davidatbu sets up the remote dev server and share my comments.

**Comment by davidatbu** [comment]
It looks like setting up the dev server will take me longer than an afternoon(sorry @monajalal !). I am going to try to locally review the PR, and then setup the dev server sometime before next week.

File: services/web/frontend/js/lda.js [hunk 1]
```diff
@@ -68,15 +68,17 @@ $(function() {
                             console.log('success in training file POST');
                             $('#tm-success-message').removeAttr('hidden');
                         },
-                        error: function (err) {
-                            console.log(err);
-                            $('#err-uploading').removeAttr('hidden');
+                        error: function (xhr, status, err) {
+                            console.log(xhr.responseText);
+                            $('#err-uploading').html(`An error occurred while uploading your file: ${JSON.parse(xhr.responseText).message}`)
```
Comment (davidatbu) [inline services/web/frontend/js/lda.js:73]: So, it turns out, there are two kinds of error messages that the server sends. The first format (which you've addressed) is:

```javascript
{
   "message": "...",
}
```
The second format is
```javascript
{
    "message": {
        "num_topics": "The number of topics must be an integer greater than 1."
    }
}
```

The second format is returned only when the error pertains specifically to one field (like the number of topics, in this example). 

Could you try to account for both types of errors here (and also on the classifiers page)?

Tbh, the possible errors the server can spew out should be documented in the backend API. I'll create an issue for that.

**Review by davidatbu — CHANGES_REQUESTED** [review]
I left a review at the line where I thought the fix might be, but I saw later that the other change in the PR also requires the same fix, sorry.

File: services/web/frontend/js/lda.js (hunk 1 above)
Comment (vidyaap) [reply services/web/frontend/js/lda.js:73]: Will do, where can I find the other formats that might get returned? Or is num_topics the only nested one?

**Comment by davidatbu** [comment]
Hi Vidya,

I will put this in the docs soon, but basically, every field in the body of
a JSON request will be validated and a possible error might be returned. My
suggested way of differentiating between the previous kinds and these kinds
of errors is checking for a ".message" property on the returned JSON, as
that is not a  field name in the body of any endpoint.

Am I making sense?

**Comment by vidyaap** [comment]
Makes sense, since they'll all have a messages property though, I'll just check whether it's an object and show whatever messages are in that object.

**Comment by davidatbu** [comment]
Yes, what you said makes much better sense actually. What I said about how to differentiate the errors makes no sense. Thanks for understanding my blabbering.

**Comment by vidyaap** [comment]
Alright, this should take care of any keys as long as it comes back in a "messages" object. Btw I didn't have any UI handling for the loopingTesting() function because I assumed those errors would get sent via email. Let me know if that is not the case!

**Comment by davidatbu** [comment]
Hey Vidya, sorry for the silence again. Thanks for this!

**Pull request merged**